### PR TITLE
CXX-2693 remove valgrind tests from Evergreen

### PR DESCRIPTION
**Pull request by rcsanchez97**
Evergreen patch build: https://spruce.mongodb.com/version/64650e92850e610c797f9013/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: .mci.yml [hunk 1]
```diff
@@ -590,7 +589,7 @@ functions:
                           echo "CRYPT_SHARED_LIB_PATH=$CRYPT_SHARED_LIB_PATH"
                       fi
 
-                      # Run tests and examples 1-by-1 with "test_params" so we can run them with valgrind.
+                      # Run tests and examples 1-by-1 with "test_params" so we can run them with asan/ubsan.
```
Comment (kevinAlbs) [inline .mci.yml:592]: I think comment is no longer relevant and can be removed.

`valgrind_test_params` included the `valgrind` command. `asan_test_params` and `ubsan_test_params` only include environment variables.

I do not think it is necessary to run the tests individually for `asan/ubsan`. I suggest removing this comment.

File: .mci.yml [hunk 2]
```diff
@@ -672,7 +671,7 @@ functions:
                           export LD_LIBRARY_PATH="$(pwd)/build/install/lib:$LD_LIBRARY_PATH"
                       fi
                   fi
-                  # The example projects never run under valgrind, since we haven't added execution
+                  # The example projects never run under asan/ubsan, since we haven't added execution
                   # logic to handle ${test_params}.
```
Comment (kevinAlbs) [inline .mci.yml:675]: This comment also no longer seems relevant. The `${test_params}` for ASAN and UBSAN only include environment variables. I suggest adding `${test_params}` below to include the environment variables:
```
                  if [[ -z "$MONGODB_API_VERSION" ]]; then
                    echo "Building example projects..."
                    ${test_params} .evergreen/build_example_projects.sh
                    echo "Building example projects... done."
                  fi
```

File: .mci.yml [hunk 3]
```diff
@@ -1374,44 +1373,6 @@ buildvariants:
...
-      display_name: "Valgrind Ubuntu 18.04 Debug (MongoDB Latest)"
-      expansions:
-          build_type: "Debug"
-          tar_options: *linux_tar_options
-          extra_path: *linux_extra_path
-          cmake_flags: *linux_cmake_flags
-          test_params: *valgrind_test_params
-          mongodb_version: *version_latest
-          disable_slow_tests: 1
-          use_mongocryptd: true  # false positives arise from the crypt_shared library
```
[the remark this answers is not in the document]
Comment (rcsanchez97) [reply .mci.yml:1387]: Interesting.  It didn't register with me that the environment was being disturbed by that.  I'll take a look.

**Comment by rcsanchez97** [comment]
The suggestion to individually export required Evergreen expansions did the trick.  Closing this PR.

**Pull request closed without merging**